terracumber/junit.py

```diff
@@ -58,3 +58,23 @@ def get_failures(self, number=-1):
                 else:
                     break
         return failures
+
+    def get_failures_saltshaker(self, number=-1):
+        """Return a list of failure messages for failed tests from Salt Shaker.
+
+        Keyword arguments:
+        number: The maximum number of messages to return, -1 for all messages
+        """
+        failures = []
+        for tfile in self.sort_test_files_by_mtime():
+            j_failures = minidom.parse(tfile).getElementsByTagName('failure')
+            for j_failure in j_failures:
+                if len(failures) < number or number == -1:
+                    failures.append("{}::{}".format(
```

Comment:
I am curious: why is this part different for shaker?

And if we need this as a different function, remember to add a unit test for it, including resources for the test.

Reply:
I had to use a different parsing function for the failures after the pytest execution in order to extract the "test_file::failing_test", which is what we need here. Otherwise I'm getting the error messages instead, which is not what I need here.

Thanks for the note about the test. Will do.

Reply:
Tests added